allauth/socialaccount/providers/agave/provider.py

```diff
@@ -12,7 +12,7 @@ def get_avatar_url(self):
 
     def to_str(self):
         dflt = super(AgaveAccount, self).to_str()
-        return self.account.extra_data.get("name", dflt)
+        return self.account.extra_data.get("email", dflt)
```

Comment:
This change assumes `email` is always present, which need not be the case. What if the scope/consent is such that this data is not present?

Reply:
The parent class's `to_str` implementation was changed as well, to retrieve the identifier from `email`, `username` and `name`. So, in this case, if `email` is not found or falsey, then the `dflt` value will probably be taken from the `name` value. I suppose I could have deleted this `to_str` method altogether and just relied on the parent class's `to_str` implementation.